syntax = "proto3";

package orders;

service orders {
    rpc get_order(GetOrderRequest) returns (OrderResponse);
    rpc create_order(CreateOrderRequest) returns (OrderResponse);
    rpc update_order(UpdateOrderRequest) returns (OrderResponse);
    rpc delete_order(DeleteOrderRequest) returns (OrderDeletedResponse);
}

message OrderDetailResponse {
    int32 id = 1;
    string product_id = 2;
    string price = 3;
    int32 quantity = 4;
}

message OrderResponse {
    int32 id = 1;
    repeated OrderDetailResponse order_details = 2;
}

message GetOrderRequest {
    int32 id = 1;
}

message CreateOrderDetailRequest {
    string product_id = 1;
    string price = 2;
    int32 quantity = 3;
}

message CreateOrderRequest {
    repeated CreateOrderDetailRequest order_details = 1;
}

message UpdateOrderDetailRequest {
    int32 id = 1;
    string product_id = 2;
    string price = 3;
    int32 quantity = 4;
}

message UpdateOrderRequest {
    int32 id = 1;
    repeated UpdateOrderDetailRequest order_details = 2;
}

message DeleteOrderRequest {
    int32 id = 1;
}

message OrderDeletedResponse {}
